test/concurrent/WorkerThreadTest.ooc

```diff
@@ -0,0 +1,48 @@
+/* This file is part of magic-sdk, an sdk for the open source programming language magic.
+ *
+ * Copyright (C) 2016 magic-lang
+ *
+ * This software may be modified and distributed under the terms
+ * of the MIT license.  See the LICENSE file for details.
+ */
+
+use concurrent
+use unit
+
+WorkerThreadTest: class extends Fixture {
+	sum: static Long = 0
+	taskUp := func {
+		for (j in 0 .. 10_000_000)
+			This sum += 1
+	}
+	taskDown := func {
+		for (j in 0 .. 10_000_000)
+			This sum -= 1
+	}
+	init: func {
+		super("WorkerThread")
+		this add("Create and wait for all", func {
+			thread := WorkerThread new()
+			thread getPromise(this taskUp) wait() . free()
+			expect(this sum, is greater than(0))
+
+			thread add(this taskUp)
+			thread add(this taskUp)
+			thread add(this taskDown)
+			thread add(this taskDown)
+			thread add(this taskUp)
```

Comment:
Maybe verify that these tasks really run in the same thread, if this is the purpose of this class.